code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +360,52 @@
 		P.fire()
 		return P
 
+
 /mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
 	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+
+/mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
+	if(environment_smash)
+		EscapeConfinement()
+
+		var/dir_to_target_cardinal = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
```

Comment:
you don't need this at all, since if `if(dir_to_target in GLOB.diagonals)` fails we know the target is either on your turf or in a cardinal direction, so we can add it to the list without checking it.